packages/grafana-ui/src/components/EmptyState/EmptyState.tsx

```diff
@@ -22,26 +22,70 @@ interface Props {
    */
   message?: string;
   /**
-   * Empty state variant. Possible values are 'search'.
+   * Which variant to use. Affects the default message and image shown.
    */
+  variant: 'initial' | 'search';
+}
+
+interface InitialVariantProps extends BaseProps {
+  message: string;
+  variant: 'initial';
+}
+
+interface SearchVariantProps extends BaseProps {
   variant: 'search';
 }
 
+type Props = InitialVariantProps | SearchVariantProps;
+
 export const EmptyState = ({
   button,
   children,
-  image = <GrotNotFound width={300} />,
-  message = t('grafana-ui.empty-state.search-message', 'No results found'),
+  image,
+  message,
   hideImage = false,
+  variant,
 }: React.PropsWithChildren<Props>) => {
+  const imageToShow = image ?? getImageForVariant(variant);
+  const messageToShow = message ?? getMessageForVariant(variant);
+
   return (
     <Box paddingY={4} gap={4} display="flex" direction="column" alignItems="center">
-      {!hideImage && image}
+      {!hideImage && imageToShow}
       <Stack direction="column" alignItems="center">
-        <Text variant="h4">{message}</Text>
+        <Text variant="h4">{messageToShow}</Text>
         {children && <Text color="secondary">{children}</Text>}
       </Stack>
       {button}
     </Box>
   );
 };
+
+function getMessageForVariant(variant: Props['variant']) {
+  switch (variant) {
+    case 'initial': {
+      return t('grafana-ui.empty-state.initial-message', "You haven't created anything yet");
```

Comment:
We might want to think about this differently - "You haven't created anything" is very editor-centric. "There's nothing here yet" might work better for all cases.

I wonder though - maybe it would be better to not have this variant and just require consumers to specify their own message? Dunno.

Reply:
yeah we can force the types to require a message for this variant, i think that's probably a good idea